codegens/curl/lib/util.js

```diff
@@ -199,5 +201,77 @@ var self = module.exports = {
         contentType: contentType
       });
     }
+  },
+
+  /**
+   * @param {Object} body
+   * @returns {boolean}
+   *
+   * Determines if a request body is actually empty.
+   * This is needed because body.isEmpty() returns false for formdata
+   * and urlencoded when they contain only disabled params which will not
+   * be a part of the curl request.
+   */
+  isBodyEmpty (body) {
+    if (!body) {
+      return true;
+    }
+
+    if (body.isEmpty()) {
+      return true;
+    }
+
+    if (body.mode === 'formdata' || body.mode === 'urlencoded') {
+      let memberCount = 0;
+      body[body.mode].members && body[body.mode].members.forEach((param) => {
```

Comment:
Add this extra condition `body[body.mode] &&` at start to avoid typeerrors in future.